torch_xla_py/xla_dist.py

```diff
@@ -402,3 +407,241 @@ def get_cluster(self):
+              ' pytorchtpudistrunner@{} --command="docker rm -f {}"').format(
+                  client_worker._zone, client_worker._hostname,
+                  self.container_name)
+    proc = subprocess.Popen(rm_cmd, stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE, shell=True)
+    self._stream_logs(proc, client_worker)
+
+  def _start_run(self, script_map):
+
+    def _gcloud_ssh(script_path, client_worker, event):
+      while event.is_set():
+        ssh_cmd = [
+            'gcloud', '-q', 'compute', 'ssh', '--internal-ip',
+            '--zone={}'.format(client_worker._zone),
+            'pytorchtpudistrunner@{}'.format(client_worker._hostname),
+            '--command=~/{}'.format(os.path.basename(script_path))
+        ]
+        proc = subprocess.Popen(ssh_cmd, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        self._stream_logs(proc, client_worker)
+      proc.kill()
+      if self.is_docker:
+        self._rm_docker_container(client_worker)
+
+    event = threading.Event()
+    event.set()
+    threads = []
+    for i, client_worker in enumerate(script_map):
+      thread = threading.Thread(
+          target=_gcloud_ssh,
+          args=(script_map[client_worker], client_worker, event,))
+      thread.daemon = True
+      thread.start()
+      threads.append(thread)
+
+    try:
+      for thread in threads:
+        thread.join()
+    except KeyboardInterrupt:
+      event.clear()
+
+    for thread in threads:
+      thread.join()
+
+  def run(self, cmd):
+    self.logger.info('Command to distribute: {}'.format(cmd),
+                     extra={'clientip': '', 'ordinal': ''})
+    self.logger.info('Cluster configuration: {}'.format(self._cluster),
+                     extra={'clientip': '', 'ordinal': ''})
+    self.is_docker = self.DOCKER_RUN in cmd
+    script_map = self._prepare_scripts(cmd)
+    self._scp_scripts(script_map)
+    self._start_run(script_map)
+
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser(
+      description="PyTorch on TPU distrubuted training",
+      epilog=("Usage example: xla_dist.py --tpus=[TPU_NAME] -- "
+              "\'conda activate pytorch-nightly && python train.py\'"))
+  parser.add_argument(
+      "positional",
+      nargs="+",
+      default="",
+      type=str,
+      help="The python command to launch training including model parameters.")
+  parser.add_argument(
+      "--tpus",
+      nargs="+",
+      default="",
+      type=str,
+      help="Name of the TPU pod, or list of single Cloud TPU devices (v*-8).")
+  parser.add_argument(
+      "--vms",
+      nargs="+",
+      default="",
+      type=str,
+      help=("(optional) List of single Compute VM instance names. "
+            "If not provided we assume usage of instance groups."))
+  FLAGS = parser.parse_args()
+  train_cmd = FLAGS.positional
+
+  # Resolve VM and TPU clusters.
+  vms = None if not FLAGS.vms else FLAGS.vms
```

Comment:
What if you leave `default` unassigned above? `FLAGS.vms` should already be `None`?